src/main/kotlin/org/radarbase/push/integration/garmin/user/GarminServiceUserRepository.kt

```diff
@@ -1,52 +1,201 @@
 package org.radarbase.push.integration.garmin.user
 
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.ObjectReader
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import io.confluent.common.config.ConfigException
+import okhttp3.*
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.RequestBody.Companion.toRequestBody
 import org.radarbase.gateway.Config
+import org.radarbase.gateway.GarminConfig
 import org.radarbase.push.integration.common.user.User
+import org.radarbase.push.integration.common.user.Users
+import org.radarcns.exception.TokenException
+import org.radarcns.oauth.OAuth2Client
+import org.slf4j.LoggerFactory
 import java.io.IOException
+import java.net.URL
+import java.time.Duration
 import java.time.Instant
+import java.util.*
 import java.util.stream.Stream
 import javax.ws.rs.NotAuthorizedException
 import javax.ws.rs.core.Context
+import kotlin.collections.HashMap
+
 
 class GarminServiceUserRepository(
-    @Context private val config: Config
+        @Context private val config: Config
 ) : GarminUserRepository(config) {
+    private val garminConfig: GarminConfig = config.pushIntegration.garmin
+    private val client: OkHttpClient? = OkHttpClient()
+    private val cachedCredentials: HashMap<String, OAuth1UserCredentials>? = HashMap<String, OAuth1UserCredentials>()
```

Comment:
Given that this class will be used in a multithreaded context, this should probably be a `ConcurrentHashMap`. It is never null, so the type should be non-null as well.